python/lsst/pipe/base/script/zip_from_graph.py

```diff
@@ -0,0 +1,121 @@
+# This file is part of pipe_base.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This software is dual licensed under the GNU General Public License and also
+# under a 3-clause BSD license. Recipients may choose which of these licenses
+# to use; please see the files gpl-3.0.txt and/or bsd_license.txt,
+# respectively.  If you choose the GPL option then the following text applies
+# (but note that there is still no warranty even if you opt for BSD instead):
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__all__ = ["zip_from_graph"]
+
+import logging
+import re
+
+from lsst.daf.butler import DatasetRef, QuantumBackedButler
+from lsst.daf.butler.utils import globToRegex
+from lsst.pipe.base import QuantumGraph
+from lsst.resources import ResourcePath
+
+_LOG = logging.getLogger(__name__)
+
+
+def zip_from_graph(
+    graph: str,
+    repo: str,
+    dest: str,
+    dataset_type: tuple[str, ...],
+) -> ResourcePath:
+    """Create Zip export file from graph outputs.
+
+    Parameters
+    ----------
+    graph : `str`
+        URI string of the quantum graph.
+    repo : `str`
+        URI to a butler configuration used to define the datastore associated
+        with the graph.
+    dest : `str`
+        Path to the destination directory for the Zip file.
+    dataset_type : `tuple` of `str`
+        Dataset type names. An empty tuple implies all dataset types.
+        Can include globs.
+
+    Returns
+    -------
+    zip_path : `lsst.resources.ResourcePath`
+        Path to the Zip file.
+    """
+    # Read whole graph into memory
+    qgraph = QuantumGraph.loadUri(graph)
+
+    # Collect output refs that could be created by this graph.
+    original_output_refs: set[DatasetRef] = set(qgraph.globalInitOutputRefs())
+    for task_def in qgraph.iterTaskGraph():
+        if refs := qgraph.initOutputRefs(task_def):
+            original_output_refs.update(refs)
+    for qnode in qgraph:
+        for otherRefs in qnode.quantum.outputs.values():
+            original_output_refs.update(otherRefs)
```

Comment:
Would it be useful to have that code as a method in the graph class?

Reply:
Probably.